plugins/module_utils/firewalld.py

```diff
@@ -8,7 +8,14 @@
 __metaclass__ = type
 
 # Imports and info for sanity checking
-from distutils.version import LooseVersion
+try:
+    # For ansible-core 2.12 or later:
+    from ansible.module_utils.compat.version import LooseVersion
+except ImportError:
+    # For distutils for ansible-core 2.11 or earlier. Note that distutils
+    #  will be removed from Python 3.12.
+    # https://www.python.org/dev/peps/pep-0632/
+    from distutils.version import LooseVersion
```

Comment:
```suggestion
    from distutils.version import LooseVersion
```
I like @felixfontein 's approach https://github.com/ansible-collections/community.general/pull/3936. In particular, the error message.

Reply:
@Andersson007 thank you for your advice! Yeah, it's better than mine. I'll modify it :)